day2/src/main.rs

```diff
@@ -0,0 +1,59 @@
+use itertools::Itertools;
+use std::time::Instant;
+
+const INPUT: &[u8] = include_bytes!("input.txt");
+
+fn parse(array: &[u8]) -> i32 {
+    if array.len() == 1 {
+        return (array[0] - b'0') as i32;
+    }
+    return (array[0] - b'0') as i32 * 10 + (array[1] - b'0') as i32;
+}
+
+fn report_is_safe(report: &Vec<i32>) -> bool {
```

Comment:
Though if you change this here you can't do stuff like `filter(report_is_safe)` any more you need to declare a closure so that Rust can do the type coercion e.g. `filter(|v| report_is_safe(&v))`

Reply:
hmm is this what you mean? [bdb75bf](https://github.com/zekehuntergreen/advent-of-code-2024/pull/1/commits/bdb75bf2dae55f525b67527990eef0fcab6c96af) ?
So still collecting a vector before passing to `report_is_safe`?

Reply:
Yep! But you can still do the `collect` in the same `map` as you were before I think

Reply:
oh yeah, oops!